subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/VersionInfo.java

```diff
@@ -15,22 +15,22 @@
  */
 package org.gradle.api.internal.artifacts.ivyservice.ivyresolve;
 
+import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.strategy.Version;
+import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.strategy.VersionParser;
+
 public class VersionInfo implements Versioned {
-    private final String version;
+    private final Version version;
 
     public VersionInfo(String version) {
-        this.version = version;
+        this.version = VersionParser.INSTANCE.transform(version);
```

Comment:
Same here, we could probably reuse the factory and reclaim a bit of memory/cpu cycles.